tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/InlinePassImpl.scala

```diff
@@ -63,6 +63,7 @@ class InlinePassImpl @Inject() (val options: PassOptions, gen: UniqueNameGenerat
       List(
           InlinerOfUserOper(defBodyMap, tracker),
           LetInExpander(tracker, keepNullary = true),
+          FoldOperLetinizer(tracker, defBodyMap, gen),
```

Comment:
Although this approach solves the problem, we should think about a more general approach, that is, operators that receive lambdas.

Reply:
Yeah, delegated this to #872 .